### ISSUE TITLE
geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by geckodriver) geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by geckodriver)

### ISSUE THREAD

**Issue by vaccarieli** [jamesmortensen/geckodriver-arm-binaries#4]
Hi i am gettimg this error when trying to run, i have tried to update my libc6 on my device but 2.31 is the latest , any solution you can help me? 

``geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by geckodriver)
geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by geckodriver)``

**Comment by taz5176** [comment]
hi
i too had the same failure, im on lenovo chromebook duet 3 (ver 103.0.5060.132). i found that by adding "deb http://ftp.de.debian.org/debian sid main" will to the sources.list will fix the issue. 

But this will break the android on my chromebook. my play store and android apps will not work after adding the source, But surprisingly, diablo immortal android app is working. i'm not sure why as i'm new too

**Comment by jamesmortensen** [comment]
@vaccarieli what OS are you using?  Is it for a VM or a container or a local install?  We use the geckodriver ARM binary in the https://github.com/seleniumhq-community/docker-seleniarm ARM ports of docker-selenium.  These container images are built on Debian using sid sources, which coincides with what @taz5176 experiences as well.

**Comment by jowerner** [comment]
Hi,

same issue here: 

```
$ geckodriver --version
geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by geckodriver)
geckodriver: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by geckodriver)
```

We are seeing this on a VM instance with the latest Debian 11 image in the Google cloud.

**Comment by jamesmortensen** [comment]
@jowerner Do you think any of these suggestions might help?  https://stackoverflow.com/q/10863613/552792

Some answers suggest you can apt-install the glibc library from the sid mirrors.

**Comment by jamesmortensen** [comment]
@jowerner you mentioned the VM is on Google Cloud.  Is it an ARM Ampere Ultra?

**Comment by jamesmortensen** [comment]
```
           echo "HOST_TRIPLE=aarch64-unknown-linux-gnu" >> $GITHUB_ENV
```
https://github.com/jamesmortensen/geckodriver-arm-binaries/blob/42266f2e94763a1f233ce94f6a0fedfdddc8969b/.github/workflows/build-and-release.yml#L27

I won't have time to try this week, but if you Change gnu to musl on this line, we can see if it statically links glibc to the geckodriver binary.  Hope this helps.

**Comment by jamesmortensen** [comment]
In other news, Mozilla is working on an official arm64 geckodriver release for 0.32.0.  If you want to see if glibc is statically linked  in that build, you can download their prerelease from here:

https://treeherder.mozilla.org/jobs?repo=try&revision=c8ab7c89b22fa372f5e0762b0267cb54f5c40b50&selectedTaskRun=DTtanCaxRZ2ILW93X4MIBQ.0

**Comment by jowerner** [comment]
Hi @jamesmortensen,
Thanks for looking into this!

> @jowerner Do you think any of these suggestions might help? https://stackoverflow.com/q/10863613/552792
>
> Some answers suggest you can apt-install the glibc library from the sid mirrors.

Thanks, have to check if that will work for us.

> @jowerner you mentioned the VM is on Google Cloud. Is it an ARM Ampere Ultra?

Yes, exactly.

> In other news, Mozilla is working on an official arm64 geckodriver release for 0.32.0. If you want to see if glibc is statically linked in that build, you can download their prerelease from here:

Tried that and that version does not show the issue! Not sure though if they statically link geckodriver or if they just depend on glibc 2.31. Anyway, their binary is smaller than the one you provide (9 vs. 18 MB).

**Comment by jamesmortensen** [comment]
That's great news. Once they release the official aarch64 geckodriver 0.32.0, I'll point people back to that and just keep this repo for historical purposes (or to build geckodriver for RasberryPi arm32 platforms).

Not sure why this one is bigger.  I built a version using cross compilation instead of emulation, and the tarball was 3.14mb instead of 4.69mb.  You can see the cross-compilation branch if interested.  At any rate, I probably won't do much further with this repo, now that Mozilla provides official support.

**Comment by jamesmortensen** [comment]
## Release Update

Mozilla has released v0.32.0 of geckodriver for linux-aarch64.  Please use their official version instead of the one here.   See [mozilla/releases](https://github.com/mozilla/geckodriver/releases) for more details.

Should you still need the unofficial one, I have built geckodriver for both linux-aarch64 and linux-armv7, and those are available on the [geckodriver-arm-binaries/releases](https://github.com/jamesmortensen/geckodriver-arm-binaries/releases) page.

I don't consider it a high priority to resolve the static glibc issue, since Mozilla now provides a solution; however, if someone has a particular need and wants to work on it and open a pull request, I'm happy to help.
